### PR TITLE
Rolling Buffer changes to handle retransmission

### PR DESCRIPTION
**Pull request by zhiyua-git**
*Issue #, if available:*
N/A
*Description
Rolling buffer that holds data and let data out of range fall off and freed.
Rtp rolling buffer to keep packets and return packet for resend if required.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/Rtcp/RollingBuffer.c [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+#include "../Include_i.h"
+
+STATUS createRollingBuffer(UINT32 capacity, FreeDataFunc freeDataFunc, PRollingBuffer* ppRollingBuffer)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PRollingBuffer pRollingBuffer = NULL;
+    CHK(capacity != 0, STATUS_INVALID_ARG);
+
+    pRollingBuffer = (PRollingBuffer) MEMALLOC(SIZEOF(RollingBuffer) + SIZEOF(UINT64) * capacity);
```
Comment (Sean-Der) [inline src/source/Rtcp/RollingBuffer.c]: Mind doing `MEMCALLOC` instead of setting 0 explicitly, just in case new members get added.

**Comment by Sean-Der** [comment]
We really need to get ASAN going, looks like we had memory corruption on this build https://travis-ci.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/jobs/261136514

Possibly unrelated!

File: src/source/Rtcp/RollingBuffer.c (hunk 1 above)
Comment (zhiyua-git) [reply src/source/Rtcp/RollingBuffer.c]: The performance is better on using MEMALLOC, given the rolling buffer could be very large, I would rather do MEMALLOC.

File: src/source/Rtcp/RollingBuffer.c [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    pRollingBuffer->dataBuffer = (PUINT64) (pRollingBuffer + SIZEOF(RollingBuffer));
+    MEMSET(pRollingBuffer->dataBuffer, 0, SIZEOF(UINT64) * pRollingBuffer->capacity);
+
+CleanUp:
+    if (STATUS_FAILED(retStatus) && pRollingBuffer != NULL) {
+        freeRollingBuffer(&pRollingBuffer);
+        pRollingBuffer = NULL;
+    }
+
+    if (pRollingBuffer != NULL) {
```
[the remark this answers is not in the document]
Comment (zhiyua-git) [reply src/source/Rtcp/RollingBuffer.c]: I should actually check for ppRollingBuffer != NULL instead. Updated.

**Pull request merged**